sklearn/cluster/tests/test_k_means.py

```diff
@@ -533,45 +511,26 @@ def test_minibatch_set_init_size():
     _check_fitted_model(mb_k_means)
 
 
-def test_k_means_copyx():
-    # Check if copy_x=False returns nearly equal X after de-centering.
+def test_kmeans_copyx():
+    # Check that copy_x=False returns nearly equal X after de-centering.
     my_X = X.copy()
     km = KMeans(copy_x=False, n_clusters=n_clusters, random_state=42)
     km.fit(my_X)
     _check_fitted_model(km)
 
-    # check if my_X is centered
-    assert_array_almost_equal(my_X, X)
-
-
-def test_k_means_non_collapsed():
-    # Check k_means with a bad initialization does not yield a singleton
-    # Starting with bad centers that are quickly ignored should not
-    # result in a repositioning of the centers to the center of mass that
-    # would lead to collapsed centers which in turns make the clustering
-    # dependent of the numerical unstabilities.
-    my_X = np.array([[1.1, 1.1], [0.9, 1.1], [1.1, 0.9], [0.9, 1.1]])
-    array_init = np.array([[1.0, 1.0], [5.0, 5.0], [-5.0, -5.0]])
-    km = KMeans(init=array_init, n_clusters=3, random_state=42, n_init=1)
-    km.fit(my_X)
-
-    # centers must not been collapsed
-    assert len(np.unique(km.labels_)) == 3
+    # check that my_X is de-centered
+    assert_allclose(my_X, X)
 
-    centers = km.cluster_centers_
-    assert np.linalg.norm(centers[0] - centers[1]) >= 0.1
-    assert np.linalg.norm(centers[0] - centers[2]) >= 0.1
-    assert np.linalg.norm(centers[1] - centers[2]) >= 0.1
 
+@pytest.mark.parametrize("Estimator", [KMeans, MiniBatchKMeans])
+def test_score_max_iter(Estimator):
+    # Check that fitting KMeans or MiniBatchKMeans with more iterations gives
+    # better score
+    X = np.random.RandomState(0).randn(100, 10)
 
-@pytest.mark.parametrize('algo', ['full', 'elkan'])
-def test_score(algo):
-    # Check that fitting k-means with multiple inits gives better score
```

Comment:
The comment does not match the code. It's actually testing that increasing max_iter gives better score.
I changed the comment. The testing of increasing n_init is covered by another already existing test.